sunpy/coordinates/metaframes.py

```diff
@@ -181,9 +182,22 @@ def __init__(self, *args, **kwargs):
         if kwargs['base'].obstime is None:
             raise ValueError("The base coordinate frame must have a defined `obstime`.")
 
+        base_frame = kwargs['base']
+        if 'duration' in kwargs:
+            duration = kwargs['duration']
+            if isinstance(duration, TimeDelta):
+                kwargs['duration'] = duration.to(u.day)
+                kwargs['rotated_time'] = base_frame.obstime + duration
+            elif isinstance(duration, u.Quantity):
+                if duration.unit != u.day:
+                    kwargs['duration'] = duration.to(u.day)
+            else:
+                raise ValueError("`duration` must be a `TimeDelta` or `Quantity` object.")
+
+        # Handle 'rotated_time' to calculate 'duration'
         if 'rotated_time' in kwargs:
             rotated_time = parse_time(kwargs['rotated_time'])
-            kwargs['duration'] = (rotated_time - kwargs['base'].obstime).to('day')
+            kwargs['duration'] = (rotated_time - base_frame.obstime).to('day')
             kwargs.pop('rotated_time')
```

Comment:
Replace this whole block with the following code:
```python
        if ('duration' in kwargs) and ('rotated_time' in kwargs):
            raise ValueError("Specify either `duration` or `rotated_time`, not both.")

        # Use rotated_time, otherwise add duration to obstime to naturally handle Quantity/TimeDelta
        rotated_time = parse_time(kwargs.pop('rotated_time', kwargs['base'].obstime + kwargs['duration']))
        # Convert back to duration
        kwargs['duration'] = (rotated_time.utc - kwargs['base'].obstime).to('day')
```

Reply:
Thank you for your review, @ayshih. I tried implementing the suggested code, but I encountered numerous errors, some of which don't seem directly related to this particular code. So I developed a different implementation that I believe achieves the same functionality as the one proposed. Do you think this is acceptable, or should I rewrite and modify the tests according to your suggested code?